fraud/bad_encoding.go

```diff
@@ -178,34 +178,39 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 }
 
 // OnBEFP listens to Bad Encoding Fraud Proof and stops services immediately if it is received.
-func OnBEFP(ctx context.Context, s Subscriber, stop func(context.Context) error) {
+func OnBEFP(ctx context.Context, s Subscriber, start, stop func(context.Context) error) error {
```

Comment:
With the suggestion above and with type param this can now become an OnFraud

Reply:
We should also consider moving this out of fraud into the node package as it is clearly now a generic lifecycle handling for services that depend on frauds.